src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,36 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+
+  const handleSelection = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterSlice.actions.setStatus(event.target.value));
```

Comment:
The `event.target.value` is of type `string`, but it should match the `Status` type used in your application. Consider casting it to `Status` or validating it before dispatching the action to ensure type safety.